brownie/project/main.py

```diff
@@ -292,30 +292,116 @@ def _compile_interfaces(self, compiled_hashes: Dict) -> None:
             self._build._add(abi)
 
     def _load_deployments(self) -> None:
-        if CONFIG.network_type != "live":
+        if CONFIG.network_type != "live" and not CONFIG.settings.get("dev_deployment_artifacts"):
             return
-        chainid = CONFIG.active_network["chainid"]
+        chainid = "dev" if CONFIG.network_type != "live" else CONFIG.active_network["chainid"]
         path = self._path.joinpath(f"build/deployments/{chainid}")
         path.mkdir(exist_ok=True)
         deployments = list(path.glob("*.json"))
         deployments.sort(key=lambda k: k.stat().st_mtime)
+        deployment_map: Dict = {}
         for build_json in deployments:
             with build_json.open() as fp:
                 build = json.load(fp)
-            if build["contractName"] not in self._containers:
+
+            contract_name = build["contractName"]
+            if contract_name not in self._containers:
                 build_json.unlink()
                 continue
             if "pcMap" in build:
                 contract = ProjectContract(self, build, build_json.stem)
             else:
                 contract = Contract(  # type: ignore
-                    build["contractName"], build_json.stem, build["abi"]
+                    contract_name, build_json.stem, build["abi"]
                 )
                 contract._project = self
-            container = self._containers[build["contractName"]]
+            container = self._containers[contract_name]
             _add_contract(contract)
             container._contracts.append(contract)
 
+            contract_address = build_json.stem
+            if chainid in deployment_map:
+                if contract_name in deployment_map[chainid] and isinstance(
+                    deployment_map[chainid][contract_name], list
+                ):
+                    deployment_map[chainid][contract_name].insert(0, contract_address)
+                else:
+                    deployment_map[chainid][contract_name] = [contract_address]
+            else:
+                deployment_map[chainid] = {contract_name: [contract_address]}
+        with self._path.joinpath("build/deployments/map.json").open("w") as fp:
+            json.dump(deployment_map, fp, sort_keys=True, indent=2, default=sorted)
+
+    def _clear_dev_deployments(self, after_block: int = 0) -> None:
+        path = self._path.joinpath(f"build/deployments/dev")
+        if path.exists():
+            deployments = list(path.glob("*.json"))
+            deployment_map = self._load_deployment_map()
+            for deployment in deployments:
```

Comment:
`deployment` and `deployments` are very similar names.  Is there a need to create `deployments`? You could iterate directly from `path.glob('*.json')`